mobile/native/src/trade/position/handler.rs

```diff
@@ -66,24 +66,28 @@ pub fn get_positions() -> Result<Vec<Position>> {
 ///
 /// If the new order submitted is an order that closes the current position, then the position will
 /// be updated to `Closing` state.
-pub fn update_position_after_order_submitted(submitted_order: Order) -> Result<()> {
-    if let Some(position) = db::get_positions()?.first() {
-        // closing the position
-        if position.direction == submitted_order.direction.opposite()
-            && position.quantity == submitted_order.quantity
-        {
-            db::update_position_state(position.contract_symbol, PositionState::Closing)?;
-            let mut position = position.clone();
-            position.position_state = PositionState::Closing;
-            event::publish(&EventInternal::PositionUpdateNotification(position));
-        } else {
-            bail!("Currently not possible to extend or reduce a position, you can only close the position with a counter-order");
-        }
+pub fn update_position_after_order_submitted(submitted_order: &Order) -> Result<()> {
+    if let Some(position) = get_position_matching_order(submitted_order) {
+        db::update_position_state(position.contract_symbol, PositionState::Closing)?;
+        let mut position = position.clone();
+        position.position_state = PositionState::Closing;
+        event::publish(&EventInternal::PositionUpdateNotification(position));
     }
-
     Ok(())
 }
 
+/// Returns the position that would be closed by submitted, if there is any
+pub fn get_position_matching_order(order: &Order) -> Result<Option<Position>> {
+    Ok(if let Some(position) = db::get_positions()?.first() {
+        // closing the position
+        ensure!(position.direction == order.direction.opposite()
+                && position.quantity == order.quantity, "Currently not possible to extend or reduce a position, you can only close the position with a counter-order");
```

Comment:
I understand that we want to check this, but for me it's unexpected that the check is happening implicitly when calling `get_position_matching_order`.

I think it would be clearer if we extracted this into a separate function `does_order_close_position` and just call it if we do have an order in the database. It's two calls rather than one, but the current behaviour is surprising to me.

Reply:
Fair enough. I guess now that you extracted this function it just feels weird that this _new_ function can error like this, whereas originally it didn't bother me. It's a minor detail, specially since as you say it was already effectively the same before your changes.

Reply:
I agree with @luckysori that `does_order_close_position` is a better name than `get_position_matching_order` - the function was newly introduced. I think it'd be great to rename it.

Reply:
@da-kami : @luckysori didn't want to rename this, we wanted to extract one more function, probably with `bool` return. I opted to not do this for now, out of fear of last-minute in-adverted changes to the logic.

Reply:
Let's fix that up when we're removing `PositionState` from being stored in the db.